lib/api/apiUtils/authorization/prepareRequestContexts.js

```diff
@@ -6,37 +6,38 @@ const RequestContext = policies.RequestContext;
  * Prepares the requestContexts array to send to Vault for authorization
  * @param {string} apiMethod - api being called
  * @param {object} request - request object
- * @param {string} locationConstraint - locationConstraint if bucket put
- * operation
  * @param {string} sourceBucket - name of sourceBucket if copy request
  * @param {string} sourceObject - name of sourceObject if copy request
  * @return {RequestContext []} array of requestContexts
  */
-export default function prepareRequestContexts(apiMethod, request,
-    locationConstraint, sourceBucket, sourceObject) {
+export default function prepareRequestContexts(apiMethod, request, sourceBucket,
+    sourceObject) {
     // if multiObjectDelete request, we want to authenticate
     // before parsing the post body and creating multiple requestContexts
     // so send null as requestContexts to Vault to avoid authorization
     // checks at this point
-    if (apiMethod === 'multiObjectDelete') {
+    //
+    // If bucketPut request, we want to authenticate in the API itself because
+    // we only have access to the locationConstraint at that point.
```

Comment:
If bucketPut request , we want to do the authorization check in the API itself (once we parse the locationConstraint from the xml body) so send null as the
requestContext to Vault so it will only do an authentication check.